Failure to compile micronaut-sql.jdbc-ucp due to a classcast exception in BeanProcessor

As part of the process of getting the modules ready for 4.0.0, we have a strange error in the `jdbc-ucp` module.

https://github.com/micronaut-projects/micronaut-sql/pull/725

The `compileJava` task fails with:

```
Caused by: java.lang.ClassCastException: class com.sun.tools.javac.code.Type$ClassType cannot be cast to class javax.lang.model.element.Element (com.sun.tools.javac.code.Type$ClassType is in module jdk.compiler of loader 'app'; javax.lang.model.element.Element is in module java.compiler of loader 'platform')
        at io.micronaut.annotation.processing.BeanDefinitionInjectProcessor.process(BeanDefinitionInjectProcessor.java:241)
        at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
        at org.gradle.api.internal.tasks.compile.processing.DynamicProcessor.process(DynamicProcessor.java:52)
        at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
        at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.access$401(TimeTrackingProcessor.java:37)
        at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:99)
        at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:96)
        at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.track(TimeTrackingProcessor.java:117)
        at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.process(TimeTrackingProcessor.java:96)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1023)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:939)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1267)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1382)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1234)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:916)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.invocationHelper(JavacTaskImpl.java:152)
```

# Steps to reproduce

- Clone https://github.com/micronaut-projects/micronaut-sql
- Check out `prepare-for-mn4`
- Switch to Java 17
- Run `./gradlew :jdbc-ucp:compileJava`

# Build scan

https://ge.micronaut.io/s/vy3rwvg7x6evi

--- a/inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaElementFactory.java
+++ b/inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaElementFactory.java
@@ -318,11 +318,11 @@ public class JavaElementFactory implements ElementFactory<Element, TypeElement, 
         TypeMirror returnType = executableElement.getReturnType();
         TypeKind returnKind = returnType.getKind();
         if (returnKind == TypeKind.ERROR) {
-            throw new PostponeToNextRoundException(returnType);
+            throw new PostponeToNextRoundException(executableElement);
         }
     }
 
-    private void failIfPostponeIsNeeded(VariableElement  variableElement) {
+    private void failIfPostponeIsNeeded(VariableElement variableElement) {
         TypeMirror type = variableElement.asType();
         if (type.getKind() == TypeKind.ERROR) {
             throw new PostponeToNextRoundException(variableElement);

